src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditionsImpl.hpp

```diff
@@ -0,0 +1,1447 @@
+        char_speeds{{get(get<::Tags::TempScalar<12, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<13, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<14, DataVector>>(buffer)),
+                     get(get<::Tags::TempScalar<15, DataVector>>(buffer))}};
+
+    // Memory allocated for return type
+    ReturnType& bc_dt_u_psi =
+        get<::Tags::Tempaa<27, VolumeDim, Frame::Inertial, DataVector>>(buffer);
+    std::fill(bc_dt_u_psi.begin(), bc_dt_u_psi.end(), 0.);
+    // Switch on prescribed boundary condition method
+    switch (Method) {
+      case VSpacetimeMetricBcMethod::Freezing:
+        return bc_dt_u_psi;
+      case VSpacetimeMetricBcMethod::ConstraintPreservingBjorhus:
+        return apply_bjorhus_constraint_preserving(
+            make_not_null(&bc_dt_u_psi), unit_interface_normal_vector,
+            three_index_constraint, char_projected_rhs_dt_u_psi, char_speeds);
+      case VSpacetimeMetricBcMethod::ConstraintPreservingDirichlet:
+        return apply_dirichlet_constraint_preserving(
+            make_not_null(&bc_dt_u_psi), lapse, shift, pi, phi);
+      case VSpacetimeMetricBcMethod::Unknown:
+      default:
+        ASSERT(false,
+               "Requested BC method fo VSpacetimeMetric not implemented!");
+    }
+    // dummy return to suppress compiler warning
+    return ReturnType{};
+  }
+
+ private:
+  static ReturnType apply_bjorhus_constraint_preserving(
+      gsl::not_null<ReturnType*> bc_dt_u_psi,
+      const tnsr::A<DataVector, VolumeDim, Frame::Inertial>&
+          unit_interface_normal_vector,
+      const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>&
+          three_index_constraint,
+      const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>&
+          char_projected_rhs_dt_u_psi,
+      const std::array<DataVector, 4>& char_speeds) noexcept;
+  static ReturnType apply_dirichlet_constraint_preserving(
+      gsl::not_null<ReturnType*> bc_dt_u_psi, const Scalar<DataVector>& lapse,
+      const tnsr::I<DataVector, VolumeDim, Frame::Inertial>& shift,
+      const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>& pi,
+      const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>& phi) noexcept;
+};
+
+template <typename ReturnType, size_t VolumeDim>
+ReturnType
+set_dt_u_psi<ReturnType, VolumeDim>::apply_bjorhus_constraint_preserving(
+    const gsl::not_null<ReturnType*> bc_dt_u_psi,
+    const tnsr::A<DataVector, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataVector, VolumeDim, Frame::Inertial>&
+        three_index_constraint,
+    const tnsr::aa<DataVector, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_u_psi,
+    const std::array<DataVector, 4>& char_speeds) noexcept {
+  ASSERT(get_size(get<0, 0>(*bc_dt_u_psi)) ==
+             get_size(get<0>(unit_interface_normal_vector)),
+         "Size of input variables and temporary memory do not match.");
```

Comment:
Print both sizes as well, instead of just saying they don't match

Reply:
included sizes in this and all other asserts